src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php

```diff
@@ -409,4 +396,47 @@ private function getErrorMessages()
             DuplicateWebserviceKeyException::class => $this->trans('This key already exists.', 'Admin.Advparameters.Notification'),
         ];
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return array<string, bool|string|null>
+     */
+    private function getWebServiceStatus(Request $request): array
+    {
+        $webserviceConfiguration = $this->get('prestashop.admin.webservice.form_data_provider')->getData();
+        $webserviceStatus = [
+            'isEnabled' => (bool) $webserviceConfiguration['enable_webservice'],
+            'isFunctional' => false,
+            'endpoint' => null,
+        ];
+
+        if ($webserviceStatus['isEnabled']) {
+            $webserviceStatus['endpoint'] = $request->getSchemeAndHttpHost() . self::WEBSERVICE_ENTRY_ENDPOINT;
+            $webserviceStatus['isFunctional'] = $this->checkWebserviceEndpoint($webserviceStatus['endpoint']);
+        }
+
+        return $webserviceStatus;
+    }
+
+    /**
+     * @param string $url
+     *
+     * @return bool
+     */
+    private function checkWebserviceEndpoint(string $url): bool
+    {
+        $client = new Client();
+        $response = $client->request('GET', $url, [
+            'http_errors' => false,
```

Comment:
```suggestion
            'http_errors' => false,
            'allow_redirects' => true,
```
Make sure it follows redirection such as http to https